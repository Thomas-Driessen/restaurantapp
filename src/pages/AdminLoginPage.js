import React from 'react';
import { observer } from 'mobx-react';
import UserStore from '../components/Admin Login/UserStore';
import LoginForm from '../components/Admin Login/LoginForm';
import SubmitButton from '../components/Admin Login/SubmitButton';
import '../App.css';

class LoginPage extends React.Component {

    componentDidMount() {
        let data = sessionStorage.getItem('sessionUserStore');
        if (data != null) {
            data = JSON.parse(data);
            UserStore.isLoggedIn = data.isLoggedIn;
            UserStore.username = data.username;
            UserStore.loading = data.loading;
        }
    }

    async doLogout() {
        UserStore.isLoggedIn = false;
        UserStore.username = '';
        sessionStorage.setItem('sessionUserStore', JSON.stringify(UserStore));

    }

    render() {

        if (UserStore.loading) {
            return (
                <div className="app">
                    <div className='container'>
                        Loading, please wait....
                    </div>
                </div>
            );
        }
        else {

            if (UserStore.isLoggedIn) {
                return (
                    <div className="app">
                        <div className='container'>
                            Welcome {UserStore.username}

                            <SubmitButton
                                text={'Log out'}
                                disabled={false}
                                onClick={() => this.doLogout()}
                            />
                        </div>
                    </div>
                );
            }

            return (
                <div className="loginPage">
                    <div className='container'>
                        <LoginForm />
                    </div>
                </div>
            );

        }
    }
}

export default observer(LoginPage);