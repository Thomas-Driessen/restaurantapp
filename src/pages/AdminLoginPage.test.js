import React from 'react';
import { render } from '@testing-library/react';
import AdminLoginPage from './AdminLoginPage';
import renderer from 'react-test-renderer';

test('Render AdminLoginPage Restaurant', () => {
    const { getByText } = render(<AdminLoginPage />);
    const linkElement = getByText(/Log in/i);
    expect(linkElement).toBeInTheDocument();
});

it('AdminLoginPage renders correctly', () => {
    const tree = renderer
        .create(<AdminLoginPage />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

afterAll(() => {
    global.gc && global.gc()
})