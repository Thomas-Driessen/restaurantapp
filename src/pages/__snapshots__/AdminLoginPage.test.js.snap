// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminLoginPage renders correctly 1`] = `
<div
  className="loginPage"
>
  <div
    className="container"
  >
    <div
      className="loginForm"
    >
      <div
        className="box"
      >
        Log in
        <div
          className="inputField"
        >
          <div
            className="MuiFormControl-root MuiTextField-root input"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="outlined-search"
              id="outlined-search-label"
            >
              Username
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                id="outlined-search"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8"
                >
                  <span>
                    Username
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="inputField"
        >
          <div
            className="MuiFormControl-root MuiTextField-root input"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
              htmlFor="password"
              id="password-label"
            >
              Password
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                disabled={false}
                id="password"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root PasswordInput-eye-10"
                  focusable="false"
                  onClick={[Function]}
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
              </div>
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="submitButton"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined btn"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Login
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
